feat: add `immerPatchState` (#24)
timdeschryver/ngrx-immer: README.md

}
```

## `immerPatchState`

Provides an Immer-version of the Signal Store's `patchState`. It adds an additional updater function
which can mutate the state.

```ts

> const UserState = signalStore(withState({
> 	id: 1,
> 	name: { firstname: 'Konrad', lastname: 'Schultz' },
> 	address: { city: 'Vienna', zip: '1010' },
> }), withComputed(({ name }) => ({ prettyName: computed(() => `${name.firstname()} ${name.lastname()}`) })));

Reviewer: Was this auto-formatted?
I do think the following reads easier:

```suggestion
const UserState = signalStore(
  withState({
    id: 1,
    name: { firstname: 'Konrad', lastname: 'Schultz' },
    address: { city: 'Vienna', zip: '1010' },
  }),
  withComputed(({ name }) => ({
    prettyName: computed(() => `${name.firstname()} ${name.lastname()}`),
  }))
);
```

const userState = new UserState();

immerPatchState(userState, (state => {
	state.name = { firstname: 'Lucy', lastname: 'Sanders' };
	state.address.zip = '1020'

> }));

Reviewer 2: In addition to formatting, I suggest providing an example that is close to real-world scenarios (without manual store instantiation), because beginners may think that this is the right/recommended way of doing things. We can also remove unnecessary parts from this example such as `withComputed`. Also, note that `UserState` is renamed to `UserStore`.

```suggestion
const UserStore = signalStore(
  withState({
    user: { firstName: 'Konrad', lastName: 'Schultz' },
    address: { city: 'Vienna', zip: '1010' },
  }),
  withMethods((store) => ({
    setLastName(lastName: string): void {
      immerPatchState(store, (state) => {
        state.user.lastName = lastName;
      });
    },
    setCity(city: string): void {
      immerPatchState(store, (state) => {
        state.address.city = city;
      });
    },
  }))
);
```

Another option is to use `signalState` instead of `signalStore` in the example:

```suggestion
const userState = signalState({
    user: { firstName: 'Konrad', lastName: 'Schultz' },
    address: { city: 'Vienna', zip: '1010' },
});

immerPatchState(userState, (state) => {
 // ...
});
```

Author: I went with `signalStore`. If you prefer `signalState`, let me know.

}
```

## `immerPatchState`

> Provides an Immer-version of the Signal Store's `patchState`. It adds an additional updater function
> which can mutate the state.

Reviewer 2: We can provide more details here:

```suggestion
Provides an Immer-version of the `patchState` function from the `@ngrx/signals` package. In addition to partial state objects and updaters that update the state immutably, it accepts updater functions that update the state in a mutable manner. Similar to `patchState`, the `immerPatchState` function can be used to update the state of both SignalStore and SignalState.
```

```ts
const UserState = signalStore(
  withState({
    id: 1,
    name: { firstname: 'Konrad', lastname: 'Schultz' },
    address: { city: 'Vienna', zip: '1010' },
  }),
  withComputed(({ name }) => ({
    prettyName: computed(() => `${name.firstname()} ${name.lastname()}`),
  }))
);

const userState = new UserState();

immerPatchState(userState, (state => {
	state.name = { firstname: 'Lucy', lastname: 'Sanders' };
	state.address.zip = '1020'
}));
```

Please note, that the updater function can only mutate a change without returning it or return an immutable 
state without mutable change.

This one is going to throw a runtime error:

```ts
// will throw because of both returning and mutable change

> immerPatchState(userState, (state) => {
> 	state.name.lastname = 'Sanders'; // mutable change
> 	return state; // returning state
> });

Reviewer 2: naming and formatting:

```suggestion
immerPatchState(userStore, (state) => {
  state.name.lastname = 'Sanders'; // mutable change
  return state; // returning state
});
```

```

> 

Reviewer 2: unnecessary empty lines can be removed for consistency
